vllm/model_executor/models/llava.py

```diff
@@ -393,7 +393,8 @@ def sample(
 
     def load_weights(self, weights: Iterable[Tuple[str, torch.Tensor]]):
         # prepare weight iterators for components
-        vit_weights, mlp_weights, llm_weights = itertools.tee(weights, 3)
+        vit_weights, mlp_weights, llm_weights, tracker = itertools.tee(weights, 4)
```

Comment:
I suggest wrapping the functionality of `itertools.tee` into another utility function.

Reply:
I have implemented the `group_weights_with_prefix` function to group weights by prefix, so that we can raise `KeyError` when weights' name missing in loaded weights instead of returning empty generator.